drivers/adc/adc_nrfx_adc.c

```diff
@@ -4,6 +4,12 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/**
+ * @file
+ * @zephyrdriver{Nordic Semiconductor nRF51x ADC}
```

Comment:
Is there any reasonable chance the documentation build infrastructure could, instead of directly processing these implementation files with doxygen, just scan the tree for these first and put them all into a file that doxygen reads instead?

Reply:
Do you mean doing a search for `@zephyrdriver`, getting a list of files, then adding those files to what doxygen processes?

I can look into that.

Reply:
No, I meant constructing a single file that had all of these markers in them, collected from the entire tree, and adding it to what doxygen processes.  IOW something that looks like:

```
/**
 * @file drivers/adc/adc_nrf_adc.c
 * @zephyrdriver{Nordic Semiconductor nRF51x ADC}
 * @dtcompatible{nordic,nrf-adc}
 */
/**
 * @file drivers/adc/adc_st_adc.c
 * @zephyrdriver{ST ADC}
 * @dtcompatible{st,adc}
 */
/**
 * @file drivers/adc/nxp_st_adc.c
 * @zephyrdriver{NXP ADC}
 * @dtcompatible{nxp,adc}
 */
```

If we still tried to run doxygen on the actual implementation file I think my concerns about exposing things formerly obscured wouldn't be addressed.

Reply:
I don't like that: line numbers for error messages would be lost.